bitnami/redis/templates/sentinel/statefulset.yaml

```diff
@@ -579,8 +583,19 @@ spec:
         {{- include "common.tplvalues.render" (dict "value" .Values.replica.sidecars "context" $) | nindent 8 }}
         {{- end }}
       {{- $needsVolumePermissions := and .Values.volumePermissions.enabled .Values.replica.persistence.enabled .Values.replica.podSecurityContext.enabled .Values.replica.containerSecurityContext.enabled }}
-      {{- if or .Values.replica.initContainers $needsVolumePermissions .Values.sysctl.enabled }}
+      {{- if or .Values.replica.initContainers $needsVolumePermissions .Values.sysctl.enabled .Values.sentinel.service.createMaster }}
       initContainers:
+        {{- if .Values.sentinel.service.createMaster }}
+        - name: kubectl
+          image: {{ template "redis.kubectl.image" . }}
+          imagePullPolicy: {{ .Values.kubectl.image.pullPolicy | quote }}
+          command: {{- toYaml .Values.kubectl.command | nindent 12 }}
+          securityContext:
+            runAsUser: 0
+          volumeMounts:
+            - name: kubectl
+              mountPath: /kubectl
```

Comment:
This is weird.. You're "copying" the `kubectl` binary on a init-container to a volume that you later on mount it on Redis container so the binary is available for Sentinel to use it when calling the "client-reconfig-script".

I wonder if it'd be simpler to run Bitnami Kubectl image as a sidecar container that listens for "relabelling orders" to do the pod's relabelling. This way the workflow would be sth like:

- Sentinel executes "client-reconfig-script" that sends a "relabelling order" to the sidecar. For instance, writing that order to a shared file.
- A script is executed on Bitnami Kubectl with an infinite loop that wait for "relabelling orders" and execute them when detected. For instance, looking for changes in a shared file.

See https://kubernetes.io/docs/tasks/access-application-cluster/communicate-containers-same-pod-shared-volume

Reply:
Thank you, seems more clean indeed, will update it

Reply:
Done